src/core/transport/secure_endpoint_transport.h

```diff
@@ -0,0 +1,67 @@
+// Copyright 2025 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GRPC_SRC_CORE_TRANSPORT_SECURE_ENDPOINT_TRANSPORT_H
```

Comment:
I think we always want to use this code for endpoint-based transports, even when InsecureCreds is used, right?  If so, I suggest removing the word "secure" from the filename.

Since there is an existing endpoint_transport.h, we could just move this code into that file.  Alternatively, we could call this something like generic_client_channel_factory.{h,cc}.

Reply:
I think endpoint_transport.h will run into some dependency issues. I'll go with the second suggestion.